File: core/op.py
from sym import *
class Op(Sym):
    tag = 'op'
    def eval(self, E):
        Sym.eval(self, E)
        if len(self.nest) == 1:
            if self.val == '+': return self.nest[0].pfxadd()
            if self.val == '-': return self.nest[0].pfxsub()
        if len(self.nest) == 2:
            if self.val == '+': return self.nest[0].add(self.nest[1])
            if self.val == '-': return self.nest[0].sub(self.nest[1])
            if self.val == '*': return self.nest[0].mul(self.nest[1])
            if self.val == '/': return self.nest[0].div(self.nest[1])
            if self.val == '^': return self.nest[0].pow(self.nest[1])
            if self.val == '?=': return self.nest[0].ass(self.nest[1])
        raise BaseException(self.dump())

File: core/op.cpp
#include "core/hpp.hpp"

Op::Op(string V):Sym("op",V){}
Sym* Op::eval(Sym*E) {
	if (val!="~") Sym::eval(E); else return nest[0];
	switch (nest.size()) {
//		case 1:
//			if (val=="+") return nest[0]->pfxadd();
//			if (val=="-") return nest[0]->pfxsub();
		case 2:
//			if (val=="+") return nest[0]->add(nest[1]);
//			if (val=="-") return nest[0]->sub(nest[1]);
//			if (val=="*") return nest[0]->mul(nest[1]);
			if (val=="/") return nest[0]->div(nest[1]);
//			if (val=="^") return nest[0]->pow(nest[1]);
//			if (val=="?=") return nest[0]->ass(nest[1]);
	}
	return this; }
